cmd/lambda/accounts/update.go

```diff
@@ -2,116 +2,53 @@ package main
-	LastModifiedOn      *int64                  `json:"lastModifiedOn"`
-	CreatedOn           *int64                  `json:"createdOn"`
-	AdminRoleArn        *string                 `json:"adminRoleArn"`        // Assumed by the master account, to manage this user account
-	PrincipalRoleArn    *string                 `json:"principalRoleArn"`    // Assumed by principal users
-	PrincipalPolicyHash *string                 `json:"principalPolicyHash"` // The policy used by the PrincipalRoleArn
-	Metadata            *map[string]interface{} `json:"metadata"`
-}
-
+// UpdateAccountByID updates an accounts information based on ID
 func UpdateAccountByID(w http.ResponseWriter, r *http.Request) {
 	accountID := mux.Vars(r)["accountId"]
 
 	// Deserialize the request JSON as an request object
-	var request *updateAccountRequest
+	d := model.Account{}
 	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&request)
+	err := decoder.Decode(&d)
 	if err != nil {
-		response.WriteAPIErrorResponse(w, http.StatusBadRequest, "ClientError", "invalid request parameters")
+		api.WriteAPIErrorResponse(w,
+			errors.NewBadRequest("invalid request parameters"))
 		return
 	}
-	request.ID = &accountID
 
-	// If the request includes a new adminRoleArn,
-	// validate that we can assume the ARN
-	if request.AdminRoleArn != nil {
-		_, err = TokenSvc.AssumeRole(&sts.AssumeRoleInput{
-			RoleArn:         request.AdminRoleArn,
-			RoleSessionName: aws.String("MasterAssumeRoleVerification"),
-		})
-
-		if err != nil {
-			response.WriteRequestValidationError(
-				w,
-				fmt.Sprintf("Unable to update account %s: "+
-					"admin role is not assumable by the master account",
-					accountID),
-			)
-			return
-		}
+	var dataSvc dataiface.AccountData
+	if err := Services.Config.GetService(&dataSvc); err != nil {
+		api.WriteAPIErrorResponse(w, err)
+		return
 	}
 
-	// Figure out which DB fields we want to update
-	// Note that not all fields are updatable
-	// eg, we don't want clients to set their own timestamps,
-	// 	or modify fields, like AccountStatus, which are  used for
-	// 	internal bookkeeping.
-	var fieldsToUpdate []string
-	accountPartial := db.Account{
-		ID: *request.ID,
-	}
-	if request.AdminRoleArn != nil {
-		fieldsToUpdate = append(fieldsToUpdate, "AdminRoleArn")
-		accountPartial.AdminRoleArn = *request.AdminRoleArn
-	}
-	if request.Metadata != nil {
-		fieldsToUpdate = append(fieldsToUpdate, "Metadata")
-		accountPartial.Metadata = *request.Metadata
-	}
-	if len(fieldsToUpdate) == 0 {
-		response.WriteRequestValidationError(
-			w,
-			fmt.Sprintf("Unable to update account %s: "+
-				"no updatable fields provided",
-				accountID),
-		)
+	var amSvc accountmanageriface.AccountManagerAPI
+	if err := Services.Config.GetService(&amSvc); err != nil {
+		api.WriteAPIErrorResponse(w, err)
 		return
 	}
 
-	// Update the DB record
-	acct, err := Dao.UpdateAccount(accountPartial, fieldsToUpdate)
+	a, err := account.GetAccountByID(accountID, dataSvc, dataSvc)
```

Comment:
Do we really need to pass in `dataSvc` twice here?

Reply:
We don't.  The question falls back do we add the smaller interfaces into the struct or do we pass them to the functions that use them.  I prefer the later but there was some concerns on the last PR that we were passing the same stuff a lot.

I could reduce this to one but it could create some weird scenarios for anyone that wanted to use this function.

I would prefer to go back to the later and pass in the interface to the functions that need them.